cla_public/templates/checker/result/_find-legal-adviser.html

```diff
@@ -3,39 +3,70 @@
 
 <section class="find-legal-adviser">
   {% if not hide_subtitle %}
-    <h2>{{ _('Find a legal adviser') }}</h2>
+    <h2 class="govuk-heading-m">{{ _('Find a legal adviser') }}</h2>
   {% endif %}
 
-  <form class="legal-adviser-search" method="GET">
-    {% call Form.fieldset(form.postcode, class_=' ', field_as_label=True) %}
-      {{ Form.text_input(form.postcode) }}
-      {% if category %}
-        <input type="hidden" name="category" value="{{ category }}" />
-      {% endif %}
-      <button class="button" type="submit">{{ _('Search') }}</button>
-    {% endcall %}
+  <form method="GET">
+    {{ Form.postcode_input(form.postcode) }}
+    {% if category %}
+      <div class="govuk-form-group">
+        <input class="govuk-input govuk-input--width-10" id="postcode" type="hidden" name="category" value="{{ category }}" />
+      </div>
+    {% endif %}
+    <button class="govuk-button" type="submit">
+      {{ _('Search') }}
+    </button>
   </form>
+  
+  {% if request and request.args and request.args.get('postcode') and request.args.get('postcode')|length > 2 and request.args.get('postcode')|length < 9 %}
+    {% set scot_postcode = "AB DD DG EH FK G1 G2 G3 G4 G5 G6 G7 G8 G9 G0 HS IV KA KW KY ML PA PH TD ZE" %}
+    {% set ni_postcode = "BT" %}
+    {% if request.args.get('postcode')[:2]|upper in scot_postcode 
+    or request.args.get('postcode')[:2]|upper in ni_postcode
+    or request.args.get('postcode')[:2]|upper == "IM" 
+    or request.args.get('postcode')[:2]|upper == "GY" 
+    or request.args.get('postcode')[:2]|upper == "JE"
+    %}
```

Comment:
This seems a bit hard to follow in the template

What do you think of having these in is_ variables (is_scottish_postcode, is_ni_postcode, is_jersey_postcode...) and moving the logic to determine them in the handler

Reply:
I agree.  I'll change this.